internal/app/machined/pkg/system/services/extension.go

```diff
@@ -102,8 +103,13 @@ func (svc *Extension) Runner(r runtime.Runtime) (runner.Runner, error) {
 	}
 
 	for _, mount := range svc.Spec.Container.Mounts {
-		if err := os.MkdirAll(mount.Source, 0o700); err != nil {
-			return nil, err
+		// if file already exists skip
```

Comment:
I would rewrite it with less nesting:

```suggestion
if _, err := os.Stat(mount.Source); err == nil {
   // already exists, skip
   continue
} else if !errors.Is(err, os.ErrNotExist) {
   return nil, err
}

if err := os.MkdirAll(mount.Source, 0o700); err != nil {
   return nil, err
}
```